### PR TITLE
(review circa mar 21 2018)

### PR DESCRIPTION
**Pull request by gbromios**

### CODE REVIEW

File: src/main/java/com/gb/cornucopia/veggie/BlockVeggieCrop.java [hunk 1]
```diff
@@ -50,110 +39,68 @@ public void setDrops(final ItemVeggieRaw raw, final ItemVeggieSeed seed) {
...
-	public java.util.List<ItemStack> getDrops(final IBlockAccess world, final BlockPos pos, final IBlockState state, final int fortune)
-	{
+	public java.util.List<ItemStack> getDrops(IBlockAccess world, BlockPos pos, IBlockState state, int fortune) {
 		// TODO: make drops better. this works for now
-		java.util.List<ItemStack> ret = new ArrayList<ItemStack>();
+		java.util.List<ItemStack> ret = new ArrayList<>();
 		ret.add(new ItemStack(this.seed));
 
-		if ((Integer)state.getValue(AGE) == MAX_AGE){
+		if (state.getValue(AGE) == getMaxAge()) {
```
[the remark this answers is not in the document]
Comment (ShaneFiliatrault) [reply src/main/java/com/gb/cornucopia/veggie/BlockVeggieCrop.java]: `getMaxAge()` is a function of the parent class `BlockCrops`, which also defines `AGE`. Keeping it stored locally is redundant unless we want to have custom crop ages.

[the remark this answers is not in the document]
Comment (Syndlig) [reply src/main/java/com/gb/cornucopia/veggie/BlockVeggieCrop.java]: `getMaxAge()` is a function of `BlockCrops` which I redefined `BlockVeggieCrop` to be a subclass of. Storing it in this class would be redundant, unless we want custom crop max ages, which could be interesting.

File: src/main/java/com/gb/cornucopia/veggie/BlockVeggieCropTall.java [hunk 2]
```diff
@@ -1,11 +1,11 @@
 package com.gb.cornucopia.veggie;
 
-import java.util.Random;
-
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
+import java.util.Random;
```
[the remark this answers is not in the document]
Comment (Syndlig) [reply src/main/java/com/gb/cornucopia/veggie/BlockVeggieCropTall.java]: Yeah, IntelliJ restructures imports to be alphabetic and separates them based on whether they're java or not. It's habitual for me to hit the "Optimize Imports" button due to my work, so I'll try to keep it to a minimum.

[the remark this answers is not in the document]
Comment (gbromios) [reply src/main/java/com/gb/cornucopia/veggie/BlockVeggieCropTall.java]: No, you should be able to use optimize imports at will; I'm just saying that there must be some way to formalize what it does that everyone can just tell their ide to do

File: src/main/java/com/gb/cornucopia/veggie/Veggie.java [hunk 3]
```diff
@@ -1,353 +1,274 @@
...
-		Blocks.PUMPKIN.setCreativeTab(CornuCopia.tabVeggies);
-		Items.PUMPKIN_SEEDS.setCreativeTab(CornuCopia.tabVeggies);
-		Items.POTATO.setCreativeTab(CornuCopia.tabVeggies);
-		Items.WHEAT.setCreativeTab(CornuCopia.tabVeggies);
-		Items.WHEAT_SEEDS.setCreativeTab(CornuCopia.tabVeggies);
+	private static final int MIN_HEIGHT = 1;
+	private static final int MAX_HEIGHT = 2;
 
-	}
+	private static final HashMap<String, Veggie> vegMap = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (Syndlig) [reply src/main/java/com/gb/cornucopia/veggie/Veggie.java]: I get what you're saying. I'll go ahead and change that stuff back to what it was before. I'll also say that the standard way Minecraft does it is bad and dumb but c'est la vie.

File: src/main/java/com/gb/cornucopia/veggie/BlockVeggieCropTall.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ShaneFiliatrault) [reply src/main/java/com/gb/cornucopia/veggie/BlockVeggieCropTall.java]: (at work so not swapping between accounts) Another thing is we'll need to make sure developers are using the same code style, i.e. tabs vs spaces, braces at end of line vs under line, etc. By default, IntelliJ uses spaces, so I had to go and edit the code style to use tabs instead.

[the remark this answers is not in the document]
Comment (gbromios) [reply src/main/java/com/gb/cornucopia/veggie/BlockVeggieCropTall.java]: Lol that's what Firefox is for

[the remark this answers is not in the document]
Comment (gbromios) [reply src/main/java/com/gb/cornucopia/veggie/BlockVeggieCropTall.java]: Lol that's what Firefox is for

[the remark this answers is not in the document]
Comment (gbromios) [reply src/main/java/com/gb/cornucopia/veggie/BlockVeggieCropTall.java]: Lol that's what Firefox is for

[the remark this answers is not in the document]
Comment (gbromios) [reply src/main/java/com/gb/cornucopia/veggie/BlockVeggieCropTall.java]: Holy shit, pardon the subway wifi triple post.

Also, isnt there some convention for like java linting that ides can use for autoformatting? Or is it just a matter of going into project preferences and getting it done that way?

If there is a way I bet @tuxide knows ?

**Pull request closed without merging**